pwndbg/dbg/lldb/__init__.py

```diff
@@ -503,7 +503,12 @@ def fetch_lazy(self) -> None:
 
     @override
     def __int__(self) -> int:
-        return self.inner.signed
+        is_num, is_sign = lldb.is_numeric_type(
+            self.inner.GetType().GetCanonicalType().GetBasicType()
+        )
+        if is_num and not is_sign:
+            return self.inner.GetValueAsUnsigned()
+        return self.inner.GetValueAsSigned()
```

Comment:
There really isn't some .GetValue that handles this by itself? Eh..

Reply:
Can you add a docstring comment to __int__ explaining its complex logic and then merge it pls